src/index.js

```diff
@@ -1,106 +1,37 @@
-        </div>
-      </div>
-      `;
-    });
+const addTodo = document.getElementById('add-btn');
+const textInput = document.getElementById('todo-input');
+// clear selected button
+const clearSelected = document.querySelector('.clearChecked');
+// reset button
+const refresh = document.querySelector('.fresher');
 
-    // Function to handle edit action
-    const handleEdit = (index) => {
-      // Get the task description element
-      const taskDescEl = document.getElementsByClassName('task-input-field')[index];
-      // Replace the task description element with an input field
-      const newInputEl = document.createElement('input');
-      newInputEl.setAttribute('type', 'text');
-      newInputEl.setAttribute('value', taskDescEl.innerText);
-      newInputEl.classList.add('edit-input');
-      taskDescEl.replaceWith(newInputEl);
-      // Focus on the input field
-      newInputEl.focus();
-      // Add event listener to the input field to handle the edit action
-      newInputEl.addEventListener('keydown', (event) => {
-        if (event.key === 'Enter') {
-          // Save the edited todo
-          todos[index].description = newInputEl.value;
-          toLocalStorage();
-          renderTodos();
-        }
-      });
-    };
-
-    // Add event listener to each edit icon
-    const editIcon = document.getElementsByClassName('edit');
-    for (let i = 0; i < editIcon.length; i += 1) {
-      editIcon[i].addEventListener('click', () => {
-        handleEdit(i);
-      });
-    }
-    // Function to handle delete action
-    const handleDelete = (index) => {
-      todos.splice(index, 1);
-      for (let i = 0; i < todos.length; i += 1) {
-        todos[i].index = i + 1;
-      }
-      toLocalStorage();
-      renderTodos();
-    };
-    // Add event listener to each delete icon
-    const deleteIcon = document.getElementsByClassName('delete');
-    for (let i = 0; i < deleteIcon.length; i += 1) {
-      deleteIcon[i].addEventListener('click', () => {
-        handleDelete(i);
-      });
-    }
-  }
-};
-
-const saveTodo = () => {
-  const todoValue = todoInput.value;
-  const isEmpty = todoValue === '';
-  if (isEmpty) {
-    todoInput.innerHTML = 'Empty todo';
-  } else {
-    const todo = {
-      description: todoValue,
-      completed: false,
-      index: todos.length + 1,
-    };
-    todos.push(todo);
-    todoInput.value = '';
-    toLocalStorage();
-    renderTodos();
+textInput.addEventListener('keydown', (event) => {
+  if (event.key === 'Enter') {
+    event.preventDefault();
+    todoAdder(todoList, textInput, renderTodo, clearSelected);
   }
-};
-
-form.addEventListener('submit', (event) => {
+});
+addTodo.addEventListener('click', (event) => {
   event.preventDefault();
-  saveTodo();
+  todoAdder(todoList, textInput, renderTodo, clearSelected);
 });
 
-window.addEventListener('load', () => {
-  if (localStorage.getItem('todos')) {
-    todos = JSON.parse(localStorage.getItem('todos'));
```

Comment:
- [ ] kindly make sure your data is preserved in the local storage, currently, when I refresh the page all data are removed

https://user-images.githubusercontent.com/65068771/230745524-72fc749d-6c10-4bc5-a5a5-95f44cddc09e.mov

Reply:
- [ ] as the previous user mentioned, when I refresh the page is empty but local storage persists, I guess it is because you are not rendering what is on the local storage, kindly fix this
![Screenshot from 2023-04-08 23-58-22](https://user-images.githubusercontent.com/66097371/230745975-ab174ae5-f378-4732-b8f8-9d683292922d.png)